spec/core/1.0.md

```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
+Receipt of this `have` message entitles the client to use all message types, properties and behavior defined in a specification of that module with the same major version number and the same, or a smaller, minor version number, except for functionality dependent on capabilities.
+The major and minor version MUST be formatted according to the `<major-version>` and `<minor-version>` grammar elements defined above.
+The major version in the `have` message MUST be one of the major versions that appeared in the original `want` message.
 
+The server MUST NOT agree to using a module if it does not support any of the major versions of that module which were offered in the `want` message.
 The server MUST NOT agree to using multiple major versions of the same module, neither in a single `have` message or across multiple `have` messages on the same server connection.
 The server MAY agree to using different major versions of the same module on different server connections.
 
-When an argument in a `want` message is identical to an earlier argument in the same `want` message, or to an argument in an earlier `want` message, the server MUST reply consistently.
-If it did not agree to the earlier argument, it MUST NOT agree to this argument, and vice versa.
+When a `want` message concerns the same module as an earlier `want` message, the server MUST reply consistently.
+If it agreed to using a particular major version of a module in an earlier `have` message, and that major version is offered again in the current `have` message, it MUST agree to this major version and offer the same minor version as in the earlier `have` message.
+If it did not agree to using a particular major version of a module in an earlier `have` message, and that major version is offered again in the current `have` message, it MUST NOT agree to this major version.
 
-The following snippet shows examples of valid and invalid pairs of `want` messages and `have` replies.
-The snippet is not a single message stream.
-Each pair of messages is to be considered individually.
-
-```vt6
-(want core1 foo2)
-(have core1.0 foo2.3)
---> valid; client can now use core1.0 module and either foo2.0, foo2.1, foo2.2 or foo2.3 module
-
-(want core1 core2)
-(have core2.1)
---> valid; client can now use either core2.0 or core2.1 module, but not core1.x modules
+If the `want` message requests usage of a capability and the server agrees to using that module, it MUST reply with a `have` message with a single argument: the capability name as accepted by `<scoped-name>`.
+Receipt of this `have` message entitles the client to use all message types, properties and behavior defined in a specification of that module with the same major version number and the same, or a smaller, minor version number, where such functionality depends on the negotiated capability.
```

Comment:
Why should the client be entitled to use everything of a module or anything from the module in the first place? Also, the second half sentence concerns capabilities, while the first one concerns modules. Split that into two sentences.